_posts/2019-07-04-exploring-nservicebus-integration-testing-options.md

```diff
@@ -0,0 +1,167 @@
+---
+typora-copy-images-to: ..\img\posts\exploring-nservicebus-integration-testing-options
+typora-root-url: ..
+layout: post
+header_image: /img/posts/exploring-nservicebus-integration-testing-options/header.jpg
+title: "Exploring NServiceBus Integration Testing options"
+author: Mauro Servienti
+synopsis: "Like when rehearsing for a show you feel the need to test the entire choreography, sometimes there is the need to test the full business scenario. When using messaging it's not easy and NServiceBus doesn't help much. Follow me in a journey that explores integration testing options with NServiceBus."
+tags:
+- NServiceBus
+- Testing
+- Integration Testing
+---
+
+NServiceBus comes with a [powerful unit testing](https://docs.particular.net/nservicebus/testing/) framework. The unit testing framework is designed to allow users to test components in isolation. Let's take, for example, the following message handler:
+
+```csharp
+public class MyReplyingHandler : IHandleMessages<MyRequest>
+{
+    public Task Handle(MyRequest message, IMessageHandlerContext context)
+    {
+        return context.Reply(new MyResponse());
+    }
+}
+```
+
+What we might want to test is that the reply occurred, this can be achieved with the following unit test:
+
+```csharp
+[Test]
+public async Task ShouldReplyWithResponseMessage()
+{
+    var handler = new MyReplyingHandler();
+    var context = new TestableMessageHandlerContext();
+
+    await handler.Handle(new MyRequest(), context)
+        .ConfigureAwait(false);
```

Comment:
make it more readable by removing ConfigureAwait, useless here (and even potentially wrong as you might want to keep any test framework specific context).